File: Components/UIComponents/Common/LabelInput.py
# Set shebang if needed
# -*- coding: utf-8 -*-
"""
Created on Tue Dec 16 17:00:16 2025

@author: mano
"""

from dash import html

def LabelInput(label, inputComponent, style='top'):

    if style == 'top':
        component = html.Div(
            children=[html.Span(label), inputComponent],
            className='LabelInput LITop'
        )
    elif style == 'side':
        component = html.Div(
            children=[html.Span(label), inputComponent],
            className='LabelInput LISide'
        )
    else:
        raise ValueError('style can only be "top" or "side"')
    return component

File: assets/JavaScript/UI/Common/LabelInput.mjs
import dash from './dash_components.mjs';

function LabelInput(label, inputComp, style='top'){
    const span = dash._html.Span({'children': label});
    if (style === 'top'){
        return dash._html.Div({
            'children': [span, inputComp],
            'className': 'LabelInput LITop'
        });
    }
    else if (style === 'side'){
        return dash._html.Div({
            'children': [span, inputComp],
            'className': 'LabelInput LISide'
        });
    }
    else {
        throw new Error('style can only be side or top');
    }
    return null;
}

export default LabelInput;